swap_meet/vendor.py

```diff
@@ -1,2 +1,158 @@
+
+        # return the list 'items_in_category'
+        # if no items match the category, return empty list
+
+        return items_in_category
+    
+    # create instance method called get_best_by_category- takes in one argument: category (a string ex: 'Clothing')
+
+    def get_best_by_category(self, category):
+
+        # create variable to hold item with highest condition in matching category, assign as None
+        # initialize highest_condition variable at 0- will use to keep track of highest condition value
+        # create variable called list_items_in_category and use get_by_category method to create list of items that are of the same category
+
+        best_item = None
+        highest_condition = 0
+        list_items_in_cat = self.get_by_category(category)
+
+        # if no items of that category in list- return None
+
+        if list_items_in_cat == []:
+            return None
+        
+        # iterate through each item in list
+
+        for item in list_items_in_cat:
+
+            # if the condition associated with that item is higher than highest_condition variable, we want to reassign the highest_condition variable and best_item variable with that condition and item
+
+            if item.condition > highest_condition:
+                highest_condition = item.condition
+                best_item = item
+        return best_item
+    
+    # create instance method called swap_best_by_category- takes 3 arguments: other_vendor, my_priority, their_priority
+    # other_vendor: who we want to trade with
+    # my_priority: category of item we want to receive
+    # their_priority: category of item they want to receive
+
+    def swap_best_by_category(self, other_vendor, my_priority, their_priority):
+
+        # we can use get_best_by_category method to assign my_item that we're trading, as our best item in the category they want to receive
+        # we can use get_best_by_category method to assign their_item they're trading as their best item in the category we want to receive
+        
+        my_item = self.get_best_by_category(their_priority)
+        their_item = other_vendor.get_best_by_category(my_priority)
+
+        # we can use swap_items method to remove the best item from each of our inventories
+        # we can use swap_items method to add the best item to each of our inventories
+        # swap_items method will return True
+
+        return self.swap_items(other_vendor, my_item, their_item)
+    
+        # swap_items method will return False if there are no matching items
+
+        # ****** original code below: ******
+    
+        # if my_item ==  None:
+        #     return False
+        # elif their_item ==  None:
+        #     return False
+        # else:
+        #     self.inventory.remove(my_item)
+        #     other_vendor.inventory.append(my_item)
+        #     other_vendor.inventory.remove(their_item)
+        #     self.inventory.append(their_item)
+        #     return True
+  
```

Comment:
It’s good practice to avoid committing extra or white space when submitting a PR as well as commented code.